File: Java/Quiz-Core/src/main/java/fr/epita/quiz/datamodel/MCQAnswer.java
package fr.epita.quiz.datamodel;

import javax.persistence.*;

@Entity
@Table(name = "MCQANSWER")
public class MCQAnswer {

    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    @Column(name = "ID")
    private Integer id;

    @OneToOne
    private User user;

    @OneToOne
    private Exam exam;

    @OneToOne
    private MCQChoice choice;

    public MCQAnswer() {

    }

    public MCQAnswer(User user, Exam exam, MCQChoice choice) {
        this.user = user;
        this.exam = exam;
        this.choice = choice;
    }

    public User getUser() {
        return user;
    }

    public Exam getExam() {
        return exam;
    }

    public MCQChoice getChoice() {
        return choice;
    }

    public Integer getId() {
        return id;
    }
}

File: Angular/src/app/datamodel/mcqanswer.ts
import { User } from "./user";
import { Exam } from "./exams";
import { MCQChoice } from "./mcqchoice";

export class MCQAnswer {
    user : User;

    exam : Exam;

    choice : MCQChoice;

    constructor(user : User, exam : Exam, choice : MCQChoice) {
        this.user = user;
        this.exam = exam;
        this.choice = choice;
    }
}